src/parquet/reader-test.cc

```diff
@@ -84,36 +84,56 @@ TEST_F(TestAllTypesPlain, TestBatchRead) {
   ASSERT_FALSE(col->HasNext());
 }
 
+void verifyScanner(Int32Scanner& scanner) {
+  int32_t val;
+  bool is_null;
+  for (size_t i = 0; i < 8; ++i) {
+    ASSERT_TRUE(scanner.HasNext());
+    ASSERT_TRUE(scanner.NextValue(&val, &is_null));
+    ASSERT_FALSE(is_null);
+  }
+  ASSERT_FALSE(scanner.HasNext());
+  ASSERT_FALSE(scanner.NextValue(&val, &is_null));
+}
```

Comment:
I have started working on PARQUET-481 with the goal to remove the reliance on the files in /data. I'll submit a separate PR for that. I can take care of that first, and then rebase this PR. But until then we need to be aware of this bug in TypedScanner.